resourcemgr/resourcemgr.c

```diff
@@ -245,7 +246,7 @@ typedef struct {
                                     //  flushed.
     UINT64 connectionId;            // Used to identify which connection owns the object,
                                     // sequence, or session.
-    void *nextEntry; // Next entry in the list; 0 to terminate list.                                
+    RESOURCE_MANAGER_ENTRY_PTR nextEntry; // Next entry in the list; 0 to terminate list.                             
 } RESOURCE_MANAGER_ENTRY;
 
 RESOURCE_MANAGER_ENTRY *entryList = 0;
```

Comment:
I’m going to back off from agreeing with you.

The main reason I created the RESOURCE_MANAGER_ENTRY_PTR structure was for debuggability.  In debugging I was having to walk through the linked list of structures.  When it was a void pointer, I had to cast every instance of this to a RESOURCE_MANAGER_ENTRY_PTR * in order to look at the data.

Outside of the structure, I don’t really care whether RESOURCE_MANAGER_ENTRY * is used or RESOURCE_MANAGER_ENTRY_PTR.  They’re the same thing and there’s really no advantage or disadvantage to using either of them or mixing them.

Forcing a consistent use seems like excessive pickiness to me.

Will Arthur
Intel Corporation
Server Security Firmware
803-216-2101

From: Arthur, Will C
Sent: Thursday, March 24, 2016 2:40 PM
To: '01org/TPM2.0-TSS' <reply@reply.github.com>; 01org/TPM2.0-TSS <TPM2.0-TSS@noreply.github.com>
Subject: RE: [TPM2.0-TSS] Debugability enhancements to RPSOURCE_MANAGER_ENTRY structure (#155)

Good point.  I can fix that.

Will Arthur
Intel Corporation
Server Security Firmware
803-216-2101

From: Philip Tricca [mailto:notifications@github.com]
Sent: Thursday, March 24, 2016 2:36 PM
To: 01org/TPM2.0-TSS <TPM2.0-TSS@noreply.github.com<mailto:TPM2.0-TSS@noreply.github.com>>
Cc: Arthur, Will C <will.c.arthur@intel.com<mailto:will.c.arthur@intel.com>>
Subject: Re: [TPM2.0-TSS] Debugability enhancements to RPSOURCE_MANAGER_ENTRY structure (#155)


In resourcemgr/resourcemgr.c<https://github.com/01org/TPM2.0-TSS/pull/155#discussion_r57366335>:

> @@ -245,7 +246,7 @@ typedef struct {

>                                      //  flushed.

>      UINT64 connectionId;            // Used to identify which connection owns the object,

>                                      // sequence, or session.

> -    void *nextEntry; // Next entry in the list; 0 to terminate list.

> +    RESOURCE_MANAGER_ENTRY_PTR nextEntry; // Next entry in the list; 0 to terminate list.

>  } RESOURCE_MANAGER_ENTRY;

>

>  RESOURCE_MANAGER_ENTRY *entryList = 0;

On line 215 above you go through the trouble of creating a type for a pointer to a RESOURCE_MANAGER_ENTRY structure. Great that you've replaced the void* in the structure with this but you've got ~20 uses of RESOURCE_MANAGER_ENTRY* in resourcemgr.c. My take on this is that we should pick one approach (RESOURCE_MANAGER_ENTRY* or RESOURCE_MANAGER_ENTRY_PTR) and use it consistently. Mixing these arbitrarily hurts readability.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/01org/TPM2.0-TSS/pull/155/files/af4ac0b43489264634d724d22170f8646611cf11#r57366335>